lib/synapse/config_generator/haproxy.rb

```diff
@@ -1319,13 +1322,38 @@ def write_config(new_config)
       if old_config == new_config
         return false
       else
-        tmp_file_path = "#{opts['config_file_path']}.tmp"
-        File.write(tmp_file_path, new_config)
-        FileUtils.mv(tmp_file_path, opts['config_file_path'])
+        staging_file_path = opts['staging_config_file_path']
```

Comment:
staging_file_path is confusing? this is actually in production environment?

Reply:
Yes, it is the production environment––but it is the "staging" configuration in that it's a candidate to be "released" after it is verified. I can change it back to `tmp_file_path` but I feel that doesn't quite capture the intent.